js/slider/Index.svelte

```diff
@@ -128,106 +162,202 @@
+		appearance: none;
+		height: var(--size-4);
+		width: var(--size-4);
+		background-color: white;
+		border-radius: 50%;
+		margin-top: -5px;
+		box-shadow:
+			0 0 0 1px rgba(247, 246, 246, 0.739),
+			1px 1px 4px rgba(0, 0, 0, 0.1);
+	}
+
+	input[type="range"]::-webkit-slider-runnable-track {
+		background: linear-gradient(
+			to right,
+			var(--color-accent) var(--range_progress),
+			var(--neutral-200) var(--range_progress)
+		);
+	}
+
+	/* firefox */
+	input[type="range"]::-moz-range-track {
+		height: var(--size-2);
+		background: var(--neutral-200);
+		border-radius: var(--radius-xl);
+	}
+
+	input[type="range"]::-moz-range-thumb {
+		appearance: none;
+		height: var(--size-4);
+		width: var(--size-4);
+		background-color: white;
+		border-radius: 50%;
+		border: none;
+		margin-top: calc(-1 * (var(--size-4) - var(--size-2)) / 2);
+		box-shadow:
+			0 0 0 1px rgba(247, 246, 246, 0.739),
+			1px 1px 4px rgba(0, 0, 0, 0.1);
+	}
+
+	input[type="range"]::-moz-range-progress {
+		height: var(--size-2);
+		background-color: var(--color-accent);
+		border-radius: var(--radius-xl);
 	}
 
 	input[type="number"] {
 		display: block;
-		position: relative;
-		outline: none !important;
-		box-shadow: var(--input-shadow);
-		border: var(--input-border-width) solid var(--input-border-color);
-		border-radius: var(--input-radius);
+		outline: none;
+		border: 1px solid var(--input-border-color);
+		border-radius: var(--radius-sm);
 		background: var(--input-background-fill);
-		padding: var(--size-2) var(--size-2);
-		height: var(--size-6);
+		padding: var(--size-2) var(--size-3);
+		height: var(--size-8);
 		color: var(--body-text-color);
 		font-size: var(--input-text-size);
 		line-height: var(--line-sm);
 		text-align: center;
+		min-width: var(--size-16);
+		transition: border-color 0.15s ease-in-out;
 	}
 
-	input:disabled {
-		-webkit-text-fill-color: var(--body-text-color);
-		-webkit-opacity: 1;
-		opacity: 1;
+	input[type="number"]:focus {
+		box-shadow: none;
+		border-width: 2px;
 	}
 
-	input[type="number"]:focus {
-		box-shadow: var(--input-shadow-focus);
-		border-color: var(--input-border-color-focus);
+	input:disabled,
+	input[disabled] {
+		-webkit-text-fill-color: var(--body-text-color);
+		opacity: 1;
+		cursor: not-allowed;
 	}
 
 	input::placeholder {
 		color: var(--input-placeholder-color);
 	}
 
-	input[disabled] {
+	input[type="range"][disabled] {
+		opacity: 0.6;
+	}
+
+	input[type="range"][disabled]::-webkit-slider-thumb,
+	input[type="range"][disabled]::-moz-range-thumb,
+	input[type="range"][disabled]::-ms-thumb,
+	input[type="range"][disabled]::-webkit-slider-thumb:hover,
+	input[type="range"][disabled]::-moz-range-thumb:hover,
+	input[type="range"][disabled]::-moz-range-track:hover {
+		background-color: var(--body-text-color-subdued);
 		cursor: not-allowed;
 	}
 
-	input[type="range"] {
-		-webkit-appearance: none;
-		appearance: none;
-		width: 100%;
-		accent-color: var(--slider-color);
-		height: 4px;
```

Comment:
height: 4px;
appears to fix it on Firefox
![image](https://github.com/user-attachments/assets/e68f294b-d3f9-4ae3-8e3f-f9ff53082759)

without:
![image](https://github.com/user-attachments/assets/56fe3332-21d0-44d3-8133-930bcf02b3d6)

So I am not getting a layout shift when it's not present. 

To be fair, a probably better option (perhaps alongside height) would be background: transparent or background that follows darkmode

Reply:
With the latest gradio 5.3.0 it seemed to be ok even on Firefox dark mode.